### PR TITLE
[Bugfix] Fix mv column type is not changed when do schema change

### PR DESCRIPTION
**Pull request by Lchangliang**
## Proposed changes

Issue Number: close #xxx
When modify column type which also belong mv, the colume type is only changed in baseIndex. The reference mv column is also old type.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Lchangliang** [comment]
run buildall

**Comment by Lchangliang** [comment]
run buildall

**Comment by dataroaring** [comment]
run buildall

**Comment by dataroaring** [comment]
run buildall

**Comment by dataroaring** [comment]
run buildall

File: regression-test/suites/schema_change_p0/schema_change_modify_mv_column_type.groovy [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+suite("schema_change_modify_mv_column_type") {
+    //test legacy planner
+
+    def dataFile = """${getS3Url()}/regression/datatypes/test_scalar_types_10w.csv"""
+
+    // define dup key table1
+    def testTable = "tbl_scalar_types_dup"
+    sql "DROP TABLE IF EXISTS ${testTable} FORCE"
+    sql """
+        CREATE TABLE IF NOT EXISTS ${testTable} (
```
Comment (starocean999) [inline regression-test/suites/schema_change_p0/schema_change_modify_mv_column_type.groovy:29]: pls add case for agg table too

File: regression-test/suites/schema_change_p0/schema_change_modify_mv_column_type.groovy [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+            if (exception != null) {
+                throw exception
+            }
+            log.info("Stream load result: ${result}".toString())
+            def json = parseJson(result)
+            assertEquals(100000, json.NumberTotalRows)
+            assertEquals(100000, json.NumberLoadedRows)
+        }
+    }
+    createMV ("""CREATE MATERIALIZED VIEW mv_${testTable} AS SELECT c_tinyint, c_bool, k1, c_smallint, c_int, c_bigint, c_largeint, c_float, c_double,  c_decimal, c_decimalv3, c_date, c_datetime, c_datev2, c_datetimev2, c_char, c_varchar, c_string FROM ${testTable} ORDER BY c_tinyint, c_bool, k1""")
```
Comment (starocean999) [inline regression-test/suites/schema_change_p0/schema_change_modify_mv_column_type.groovy:71]: pls add some agg function in mv to test if it works.

**Comment by Lchangliang** [comment]
run buildall

**Comment by Lchangliang** [comment]
run buildall

**Comment by Lchangliang** [comment]
run buildall

**Comment by Lchangliang** [comment]
run feut

**Comment by Lchangliang** [comment]
run feut

**Review by starocean999 — DISMISSED** [review]

**Review by luwei16 — APPROVED** [review]

**Comment by Lchangliang** [comment]
run buildall

**Review by luwei16 — APPROVED** [review]

**Review by starocean999 — APPROVED** [review]

**Pull request merged**